pythonfile/flask_api.py

```diff
@@ -0,0 +1,87 @@
+
+from flask import Flask, request, jsonify
+from pymongo import MongoClient
+from manifest_validator import ManifestValidator
+import time, json5, uuid, json
+from first import jsonstring
+import elastic
+from elasticsearch import Elasticsearch
+
+client=MongoClient('mongodb:27017')
+db=client.knexDBmh1
+coll=db.projects
+
+es = Elasticsearch(['http://elasticsearch:9200'])
+
+
+app=Flask(__name__)
+
+
+
+_id=uuid.uuid4()
+
+
+# add project to test with
+@app.before_first_request
+def create_project():
+    
+    jsonstring['date_creation']=time.strftime("%Y-%m-%d")
+    jsonstring['date_update']=time.strftime("%Y-%m-%d")
+    json5.dumps(jsonstring)
+    schema = open("manifest_schema.json")
+    validator = ManifestValidator(schema)
+    r = validator.validate_manifest(jsonstring)
+
+    if r == None:
+        
+        jsonstring['_id']=_id
+        coll.insert(jsonstring)
+        elastic.store_json("test", "projects", jsonstring)
+
+    
+@app.route('/', methods=['GET','POST'])
+def index():
+    return jsonify(coll.find_one(_id))
+
+# receive manifest as a jsonstring
+# {'manifest':manifest}
+@app.route('/add_project')
+def add_project():
+
+    manifest=request.json['manifest']
+    manifest['date_creation']=time.strftime("%Y-%m-%d")
+    manifest['date_update']=time.strftime("%Y-%m-%d")
+    json5.dumps(manifest)
+    schema = open("manifest_schema.json")
+    validator = ManifestValidator(schema)
+    r = validator.validate_manifest(manifest)

```

Comment:
What is `r`? If it can contain an error, please write it to `stderr` for the moment.